src/tests/components/LocationDetails.test.js

```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import LocationDetails from "../../components/LocationDetails";
+
+describe("LocationDetails", () => {
+  it("renders the correct city and location props", () => {
+    const { getByText } = render(
+      <LocationDetails city="Manchester" country="UK" />
```

Comment:
This test will not load will fail as the errorMessage prop is required but missing
```suggestion
      <LocationDetails city="Manchester" country="UK" errorMessage="" />
```
Note that by passing an empty error message in the component will load with the town/country details and the test will pass as a result. You could add another test here, to check if the error message displays:
```
it("renders the error correctly", () => {
    const { getByText } = render(
      <LocationDetails
        city="Manchester"
        country="UK"
        errorMessage="test error"
      />
    );
    expect(getByText("test error")).toBeInTheDocument();
  });
```